In [20]:
import requests
import os
api_key = os.environ['RFID_APP_KEY']
auth_headers = {'X-ZUMO-APPLICATION': os.environ['RFID_APP_KEY']}

In [3]:
list_of_users = requests.get('https://gwsmobileservice.azure-mobile.net/api/getusers' , headers={'X-ZUMO-APPLICATION': api_key })

In [4]:
print(list_of_users)
list_of_users.json()

<Response [200]>


[{'FirstName': 'Eric',
  'LastName': 'Cartman',
  'RFId': 12345,
  'Status': 'ACTIVE',
  '__deleted': False,
  'id': 1},
 {'FirstName': 'Stan',
  'LastName': 'Marsh',
  'RFId': 12346,
  'Status': 'ACTIVE',
  '__deleted': False,
  'id': 2},
 {'FirstName': 'Kyle',
  'LastName': 'BRoflovski',
  'RFId': 12347,
  'Status': 'ACTIVE',
  '__deleted': False,
  'id': 3},
 {'FirstName': 'Kenny',
  'LastName': 'McCormick',
  'RFId': 12348,
  'Status': 'ACTIVE',
  '__deleted': False,
  'id': 4},
 {'FirstName': 'Butters',
  'LastName': 'Stoch',
  'RFId': 12349,
  'Status': 'ACTIVE',
  '__deleted': False,
  'id': 5},
 {'FirstName': 'Pip',
  'LastName': 'Pirrup',
  'RFId': 12350,
  'Status': 'INACTIVE',
  '__deleted': False,
  'id': 6},
 {'FirstName': 'Alex',
  'LastName': 'Glick',
  'RFId': 12351,
  'Status': 'INACTIVE',
  '__deleted': False,
  'id': 7},
 {'FirstName': 'Chris',
  'LastName': 'Donnely',
  'RFId': 12352,
  'Status': 'INACTIVE',
  '__deleted': False,
  'id': 8},
 {'FirstName': 'Clyde',


In [14]:
ind_user = requests.get('http://gwsmobileservice.azure-mobile.net/api/getuserinfo?rfid=12350' 
                        , headers={'X-ZUMO-APPLICATION': api_key })

In [15]:
ind_user

<Response [200]>

In [18]:
ind_user.json()[0]['FirstName']

'Pip'

In [25]:
bikes = requests.get('http://gwsportal.azurewebsites.net/api/bikelogsapi')

In [28]:
from tkinter import *
import time
import subprocess
import re
auth_headers = {'X-ZUMO-APPLICATION': os.environ['RFID_APP_KEY']}

In [27]:
class Application(Frame):
    auth_headers = {'X-ZUMO-APPLICATION': os.environ['RFID_APP_KEY']}
    def bike(self, bike):
        print("This is Bike", bike)
        self.checkout = Button(self, text = "Checkout", command = lambda: self.checkoutbike(bike))
        self.checkout.pack()
        self.bike1.destroy()
        self.bike2.destroy()
    
    def checkoutbike(self, bike):
        allowed = False
        print("Checking out bike", bike)
        self.entrythingy = Entry()
        self.entrythingy.pack()
        self.checkout.destroy()
        # here is the application variable
        self.contents = StringVar()
        # set it to some value
        self.contents.set("Scan your RFID")
        # tell the entry widget to watch this variable
        self.entrythingy["textvariable"] = self.contents

#         p = subprocess.Popen('./test.sh', stdout=subprocess.PIPE, stderr=subprocess.PIPE)
#         time.sleep(10)
#         output = str(p.stdout.readline())
        output = input()
        code = re.findall('\d+', output)[0]
        auth = requests.get('http://gwsmobileservice.azure-mobile.net/api/getuserinfo?rfid='+code, headers=auth_headers)
        authjson = auth.json()[0]
        if 'FirstName' in authjson:
            allowed = True
            print("Authenticated")
        else:
            print("Not Authenticated")
            allowed = False
        if allowed:
            print("Open the Lock")
            root.destroy()
    
    def createWidgets(self):
        self.bike1 = Button(self, text = "Bike 1", command = lambda: self.bike(1))

        self.bike1.pack({"side": "left"})
        
        self.bike2 = Button(self, text = "Bike 2", command = lambda: self.bike(2))

        self.bike2.pack({"side": "left"})


    def __init__(self, master=None):
        Frame.__init__(self, master)
        self.pack()
        self.createWidgets()

root = Tk()
app = Application(master=root)
app.mainloop()
root.destroy()

This is Bike 1
Checking out bike 1
12353
Authenticated
Open the Lock


TclError: can't invoke "destroy" command:  application has been destroyed

In [ ]:
print(auth)

In [53]:
code = "12351"
auth = requests.get('http://gwsportal.azurewebsites.net/api/usersapi?id='+code)
if 'FirstName' in auth.json():
    print(auth.json()['FirstName'])
    allowed = True
    print("Authenticated")
else:
    print("Not Authenticated")
    allowed = False
if allowed:
    # Opens lock
    # Puts new bike log
    pass

Alex
Authenticated


In [4]:
test = requests.get('https://gwsmobileservice.azure-mobile.net/api/photocount', headers={'X-ZUMO-APPLICATION': api_key })

In [5]:
test.json()

{'code': 401, 'error': 'Error: Unauthorized'}

Bike Lock Table Names
Users - Represents GWS employees
                id                            int  Primary Key
                RFId                       int
                FristName           nvarchar(max)
                LastName            nvarchar(max)
                Status                   nvarchar(max)
               
Bikes - List of bikes available
                id                            int           Primary Key
                BikeName           nvarchar(max)
                SerialNo               nvarchar(max)
                Status                   nvarchar(max)
               
BikeLogs - Log file for activity on each bike
                id                                            int Primary Key
                UserId                                  int Foreign Key to User table (= id)
                BikeId                                   int           Foreign Key to Bikes table (= id)
                TransactionDate               datetime
                TransactionType               nvarchar(max)
               
Trail Camera Table Names
Photos - Link between Blob storage and trail camera
                id                            int Primary Key
                DeviceId              int           Foreign Key to TrailCams table (= id)
                PhotoName       nvarchar(max)
                PhotoUrl              nvarchar(max) URL generated when uploading photo to BLOB storage
               
Temperatures - Stores temperature uploaded periodically from camera
                id                                            int           Primary Key
                DeviceId                              int Foreign Key to TrailCams table (= id)
                TransactionDate               datetime
                Reading                                decimal(18,2)
               
TrailCams - List of available trail cameras
                id                            int Primary Key
                DeviceId              nvarchar(max)
                Location               nvarchar(max)
 
This URL  gives a good introduction to Azure mobile services  against a SQL Server database
https://msdn.microsoft.com/en-us/library/azure/jj710104.aspx
 
 
Our mobile service name is:  gwsmobileservice
URL is:  https://gwsmobileservice.azure-mobile.net/
Application Key is: (STORED IN ENVIORMENT VARIABLE as RFID_APP_KEY)
 
The Application Key needs to be passed in using the X-ZUMO-APPLICATION setting in the HTTP Header.
 
Retrieving all records in a table:
HTTP Type: GET
Request Header: X-ZUMO-APPLICATION = application key
URL: https://gwsmobileservice.azure-mobile.net/tables/<table name>
 
Example:  https://gwsmobileservice.azure-mobile.net/tables/Users
 
Returns an array of JSON objects
 
Retrieving a single record using primary key:
HTTP Type: GET
Request Header: X-ZUMO-APPLICATION = application key
URL: https://gwsmobileservice.azure-mobile.net/tables/<table name>/<item_id>
 
Example: https://gwsmobileservice.azure-mobile.net/tables/Users/1
 
Returns a single JSON objects
{
                "id": 1,
                "RFId": 12345,
                "FirstName": "Eric",
                "LastName": "Cartman",
                "Status": "ACTIVE"
}
 
Inserting a new record:
HTTP Type: POST
Request Header: X-ZUMO-APPLICATION  = application key
                                                                Content-Type = application/json
URL: https://gwsmobileservice.azure-mobile.net/tables/<table name>
Body: JSON object containing new record.
 
Example: https://gwsmobileservice.azure-mobile.net/tables/Users
{
                "RFId": 99999,
                "FirstName": "Randy",
                "LastName": "Marsh",
                "Status": "ACTIVE"
}
 
Returns a JSON object containing the new record.
{
                "id": 10,
                "RFId": 99999,
                "FirstName": "Randy",
                "LastName": "Marsh",
                "Status": "ACTIVE"
}
 
 
Updating an existing record:
HTTP Type: Patch
Request Header: X-ZUMO-APPLICATION = application key
                                                                Content-Type = application/json
URL: https://gwsmobileservice.azure-mobile.net/tables/<table name>/<item_id>
Body: JSON object containing the column(s) to be updated
 
Example: https://gwsmobileservice.azure-mobile.net/tables/Users/1
{ "Status": "INACTIVE"}
 
Returns JSON object representing the updated item.
{
                "id": 1,
                "RFId": 12345,
                "FirstName": "Eric",
                "LastName": "Cartman",
                "Status": "INACTIVE"
}
 
If you want to look up a User based on the RFID, then you need to use the $filter parameter.
HTTP Type: GET
Request Header: X-ZUMO-APPLICATION = application key
URL: https://gwsmobileservice.azure-mobile.net/tables/<table name>?$filter=(<filter>)
 
Example:  To get the record with an RFID equal to 12345, you would pass in a filter of “RFID eq 12345”.  Keep in mind that the filter must be URL encoded. Meaning that you special character (space, #, *, etc.) are replaced by the ASCII equivalent.  So the URL would be:
 
https://gwsmobileservice.azure-mobile.net/tables/Users?$filter=(RFId%20eq%2012345)
 
I’m not sure how you encode using python.  I’m assuming there is library that will do it.  But in JavaScript you do:
 
var URL = “https://gwsmobileservice.azure-mobile.net/tables/Users?filter=”;
var filter = encodeURIComponent(“RFId eq 12345”);
URL = URL + filter;
 